lib/env-bundler.js

```diff
@@ -715,30 +721,44 @@ const defaultComponentDescriptorProcessor = function(type, location, code, compo
     });
 }
 
-const assetComponentDescriptorProcessor = function(type, location, code, components) {
+const assetComponentDescriptorProcessor = async function(type, location, code, components) {
   const typePlural = typeToPlural(type);
   const codeExtractor = componentCodeExtractors[type];
 
+  const token = await getToken();
+
   return Promise.all(components.map(async (component) => {
       const code = canonizeCode(codeExtractor(component));
       console.log(`Processing ${type}: ${code}`);
 
-      const split = coreBaseApi.replace('//', '/').split('/');
-      const url = split[0] + '//' + split[1] + component.path;
+      const isProtected = component.group !== 'free';
```

Comment:
Independently if it is protected or not, I think you should apply the same way of fetching cms assets through the filebrowser api.
This will make this complex code a lot simpler.

Reply:
I'm not sure if I understand this correctly. Are you saying to remove `isProtected`? Could you please elaborate?

Reply:
Sorry, let me explain a bit better.

From my understanding it looks like we are downloading files from two different manners:
 1. public files are downloaded using the `path` property which points to portalui public url.
 2. protected files are downloaded using the `filename` property, which use the new logic to download from the `fileBrowswer` API.
 
My proposal is to only use the second approach and always download from the `fileBrowser` API independently if it is a `public` or `protected` asset.

Reply:
It is already doing that -- fetching the files (public or protected) via `fileBrowser` API. It doesn't fetch anymore from the portalui url. The `fileBrowser` API endpoint being consumed requires 2 query params: `protectedFolder=[true|false]` and `currentPath=[file path relative to public/protected root folder]`, which is set properly based on the `isProtected` boolean.